tests/core/opcodes/test_opcodes.py

```diff
@@ -123,6 +123,32 @@ def test_mul(vm_class, val1, val2, expected):
     assert result == expected
 
 
+@pytest.mark.parametrize(
+    'vm_class, base, exponent, expected',
+    (
+        (ByzantiumVM, 0, 1, 0,),
+        (ByzantiumVM, 0, 0, 1,),
+        (SpuriousDragonVM, 0, 1, 0,),
+        (SpuriousDragonVM, 0, 0, 1,),
+        (TangerineWhistleVM, 0, 1, 0,),
+        (TangerineWhistleVM, 0, 0, 1,),
+        (HomesteadVM, 0, 1, 0,),
+        (HomesteadVM, 0, 0, 1,),
+        (FrontierVM, 0, 1, 0,),
+        (FrontierVM, 0, 0, 1,),
```

Comment:
Side note: looks like it would be valuable to have a parametrized fixture in conftest that returns all mainnet VMs. Then this would reduce to two cases, and we wouldn't have to remember to update the test on every new fork, like:

```py
@pytest.mark.parametrize(
    'base, exponent, expected',
    (
        (0, 1, 0,),
        (0, 0, 1,),
    )
)
def test_exp(mainnet_vm, base, exponent, expected):
    computation = prepare_computation(mainnet_vm)
    computation.stack_push(exponent)
    computation.stack_push(base)
    computation.opcodes[opcode_values.EXP](computation)
    result = computation.stack_pop(type_hint=constants.UINT256)
    assert result == expected
```

Reply:
Lets wait for the chain-builder to land as it will make this a lot easier to do

Reply:
chain builder has landed so this should be pretty easy.